near-contract-standards/src/non_fungible_token/payout/payout_impl.rs

```diff
@@ -0,0 +1,75 @@
+use super::NonFungibleTokenPayout;
+use crate::non_fungible_token::core::NonFungibleTokenCore;
+use crate::non_fungible_token::payout::*;
+use crate::non_fungible_token::NonFungibleToken;
+use near_sdk::{assert_one_yocto, env};
+use near_sdk::{require, AccountId, Balance, IntoStorageKey};
+use std::collections::HashMap;
+impl Royalties {
+    pub fn new<S>(key_prefix: S) -> Self
+    where
+        S: IntoStorageKey,
+    {
+        let temp_accounts: HashMap<AccountId, BasisPoint> = HashMap::new();
+        let this =
+            Self { key_prefix: key_prefix.into_storage_key(), accounts: temp_accounts, percent: 0 };
+        this.validate();
+        this
+    }
+    pub(crate) fn validate(&self) {
+        require!(self.percent <= 100, "royalty percent must be between 0 - 100");
+        require!(
+            self.accounts.len() <= 10,
+            "can only have a maximum of 10 accounts spliting royalties"
+        );
+        let mut total: BasisPoint = 0;
+        self.accounts.iter().for_each(|(_, percent)| {
+            require!(*percent <= 100, "each royalty should be less than 100");
+            total += percent;
+        });
+        require!(total <= 100, "total percent of each royalty split  must be less than 100")
```

Comment:
It's not so clear for me what's the invariant here. Before reading this line I thought that `self.percent` should be equal to the `total`. Could you explain some more please to make sure I understand?

Reply:
Ok thanks I think understand it better now. But shouldn't the sum of the percentages that each user gets be less than equal the total percent of incoming balance used for royalties? In other words, shouldn't the invariant be `require!(total <= self.percent` instead of `require!(total <= 100)`?

Reply:
It's not how it works. First we take percent of the total balance. Then we distribute that percent to the accounts at their own specified percent. 

E.g. `incoming balance 100; total sum to distribute - 10% of incoming balance = 10; each account gets their share according to their individual percentage.`

Reply:
Or at least that is how it's currently done.

Reply:
Thanks, understand it now 👍